sql: fix flaky ShowSessions and disable TestColumnConversions/TIME-TIMETZ

jordanlewis opened the pull request:
-  Disables TIME/TIMETZ conversions until #26074 #26076 #26077 #26078 are addressed.

- For TestShowSessions: Needed to disable a couple of features (event logging, migrations,
  automatic upgrades) that cause automatic internal queries to be executed
  that get in the way of the expected output to show sessions.

Fixes #25331.
Fixes #25733.

jordanlewis committed:
sql: fix flaky ShowSessions

Needed to disable a couple of features (event logging, migrations,
automatic upgrades) that cause automatic internal queries to be executed
that get in the way of the expected output to show sessions.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26117)
<!-- Reviewable:end -->

knz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26117#-:-LDQ_lp7CeK827y7Yx46:bnfp4nl)*
<!-- Sent from Reviewable.io -->

knz committed:
sql: disable TIME/TIMETZ column type conversions

These conversions currently cause flaky tests. Also
the type TIMETZ is broken anyway so I'm not sure the conversion
code was actually sound.

Release note: none

knz commented:
I am hijacking this PR by adding a skip for TestColumnConversions in a separate commit.

Merging this as it is breaking all tests on `master`.

bors r+

knz commented:
merge failed due to #25870.

bors r+

knz commented:
ditto

bors r+

andreimatei commented:
I think https://github.com/cockroachdb/cockroach/pull/25673 is the better fix for `TestShowSessions`, no?
